cpp/unreal_plugins/WheeledVehicle/Source/WheeledVehicle/VehiclePawn.cpp

```diff
@@ -0,0 +1,177 @@
+//
+// Copyright(c) 2022 Intel. Licensed under the MIT License <http://opensource.org/licenses/MIT>.
+//
+
+#include "WheeledVehicle/VehiclePawn.h"
+
+#include <string>
+
+#include <Eigen/Dense>
+
+#include <Animation/AnimInstance.h>
+#include <Camera/CameraComponent.h>
+#include <Components/BoxComponent.h>
+#include <Components/InputComponent.h>
+#include <Components/SkeletalMeshComponent.h>
+#include <Engine/CollisionProfile.h>
+#include <UObject/ConstructorHelpers.h>
+
+#include "CoreUtils/Assert.h"
+#include "CoreUtils/Config.h"
+#include "CoreUtils/Log.h"
+#include "CoreUtils/Unreal.h"
+#include "WheeledVehicle/VehicleMovementComponent.h"
+
+AVehiclePawn::AVehiclePawn(const FObjectInitializer& object_initializer) : APawn(object_initializer)
+{
+    SP_LOG_CURRENT_FUNCTION();
+
+    if (!Config::s_initialized_) {
+        return;
+    }
+
+    ConstructorHelpers::FObjectFinder<USkeletalMesh> skeletal_mesh(*Unreal::toFString(Config::get<std::string>("WHEELED_VEHICLE.VEHICLE_PAWN.SKELETAL_MESH")));
+    SP_ASSERT(skeletal_mesh.Succeeded());
+
+    ConstructorHelpers::FClassFinder<UAnimInstance> anim_instance(*Unreal::toFString(Config::get<std::string>("WHEELED_VEHICLE.VEHICLE_PAWN.ANIM_INSTANCE")));
+    SP_ASSERT(anim_instance.Succeeded());
+
+    skeletal_mesh_component_ = CreateDefaultSubobject<USkeletalMeshComponent>(TEXT("AVehiclePawn::skeletal_mesh_component_"));
+    SP_ASSERT(skeletal_mesh_component_);
+    skeletal_mesh_component_->SetSkeletalMesh(skeletal_mesh.Object);
+    skeletal_mesh_component_->SetAnimClass(anim_instance.Class);
+    skeletal_mesh_component_->SetCollisionProfileName(UCollisionProfile::Vehicle_ProfileName);
+    skeletal_mesh_component_->BodyInstance.bSimulatePhysics = true;
+    skeletal_mesh_component_->BodyInstance.bNotifyRigidBodyCollision = true;
+    skeletal_mesh_component_->BodyInstance.bUseCCD = true;
+    skeletal_mesh_component_->bBlendPhysics = true;
+    skeletal_mesh_component_->SetGenerateOverlapEvents(true);
+    skeletal_mesh_component_->SetCanEverAffectNavigation(false);
+
+    RootComponent = skeletal_mesh_component_;
+
+    // Setup vehicle movement
+    vehicle_movement_component_ = CreateDefaultSubobject<UVehicleMovementComponent>(TEXT("AVehiclePawn::vehicle_movement_component_"));
+    SP_ASSERT(vehicle_movement_component_);
+    vehicle_movement_component_->SetIsReplicated(true); // Enable replication by default
+    vehicle_movement_component_->UpdatedComponent = skeletal_mesh_component_;
+    // this ensures that the body doesn't ever sleep. Need this to bypass a Chaos bug that doesn't take torque inputs to wheels into consideration for determining the sleep state of the body.
+    vehicle_movement_component_->SleepThreshold = 0;
```

Comment:
`UpdatedComponent` needs to be set at the pawn level. But why not set `SleepThreshold` in the constructor? Currently this is a brittle design because it is easy for a user to construct a `WheeledVehicleMovementComponent` and use it incorrectly because the user might forget to set `SleepThreshold`.

Reply:
Agreed, moved it to `UVehicleMovementComponent's` constructor.